corehq/apps/api/resources/v0_5.py

```diff
@@ -311,7 +316,10 @@ def obj_create(self, bundle, **kwargs):
                 created_via=USER_CHANGE_VIA_API,
                 email=bundle.data.get('email', '').lower(),
             )
-            del bundle.data['password']
+            # username cannot be updated
+            bundle.data.pop('username', None)
+            # password was just set
+            bundle.data.pop('password', None)
```

Comment:
just curious about this change. Using pop is better than del?

Reply:
I prefer `pop` since you can pass a default value to avoid raising an exception if the key does not exist. With `del` you have to guard against a `KeyError` exception.